### DISCUSSION TITLE
a few small tools i made for AIR

### DISCUSSION THREAD

**Discussion by dyxribo** [airsdk/Adobe-Runtime-Support#3218] [Show and tell]
hey all :) i've been working on some projects in AIR lately and wrote some helper tools. I don't know how many people are seeing these and actually using them but i figure it can't hurt to share what i've made anyway! but, if anyone does find any of these useful, i'd love to know!

first up, a mini terminal for on-the-fly debugging (especially useful for game development). i call it Termini. check out its features on the repo page, hopefully they'll be cool enough for you to give it a try?

https://github.com/dyxribo/termini_as3

![327292589-e7da59bd-e100-4bd8-85f9-2821e8d99b1a](https://github.com/airsdk/Adobe-Runtime-Support/assets/6477128/fea44e42-c8e5-4281-b543-a95649e1d9f3)

second up, a logging library modeled after Log4j, appropriately named Log4air!

https://github.com/dyxribo/log4air

it uses JSON for a configuration file and currently requires it to be named "log4air_config.json" in the bin folder of the application. planning on making this modular soon. It has support for custom Appenders, Loggers, & Layouts! next item on the list though is allowing multiple appender references per logger.

last but not least, a humble color library. i recently answered a question on stack overflow about easy color references, and apparently there aren't a lot of color libraries for as3 that are easily locatable? figured i would just post the one i made some time ago. i call it `Colors!`

https://github.com/dyxribo/as3_colors

it makes use of an RGBA class (and has a Color class for predefined RGBA colors) and has a variety of color manipulation methods built into each RGBA color. check out the repo for more info!

that's pretty much it for now. I have some other things that just aren't ready for general use yet (or they're under refactoring at the moment) but i may post some more stuff soon. would love to get some feedback, but happy to just put my code out there :)

long live as3!!

**Comment by Devin-Licastro** [comment]
Woo, these look awesome! Thanks for sharing!

**Reply by dyxribo** [reply]
happy to share, and thanks a ton, i appreciate it! 😊

**Comment by marchbold** [comment]
Great work!!

**Reply by dyxribo** [reply]
thank you so much!! 😄

**Comment by Michoko92** [comment]
Thank you so much for sharing, it's always nice to see people of this community helping each other. Cheers!

**Reply by dyxribo** [reply]
of course, i'm just glad to see that there are still people here to share it with. i love the progress that AIR is making, and can only hope that it can regain some of that love from the flash days!
